makerpm.pl

```diff
@@ -88,6 +88,7 @@
 Provides: %{name}-%{version} perl(WebminCore)
 Requires(pre): /usr/bin/perl
 Requires: /bin/sh /usr/bin/perl /bin/rm perl(lib) perl(open) perl(Net::SSLeay) perl(Time::Local) perl(Encode::Detect) perl(Data::Dumper) perl(File::Path) perl(File::Basename) perl(Digest::SHA) perl(Digest::MD5) openssl unzip tar
+Recommends: perl(DateTime) perl(DateTime::TimeZone) perl(DateTime::Locale) perl(Time::Piece)
```

Comment:
Similarly, do all RPM versions support `Recommends` ?

Reply:
Yes, it is. It worked for me on all CentOS versions, in including 7 and all newer distros like Alma and Rocky 8-9. It also worked on openSUSE as well.

Reply:
I would probably not use Recommends in RPMs yet. It was not in CentOS 7 version of RPM last time I checked. I guess if it is now, it is maybe OK to start using it...but, I'd want to really be sure of that.

Reply:
Yeah, we get a surprising number of people who still want to install Webmin on CentOS 6 (or older!)

Reply:
https://stackoverflow.com/questions/52816024/getting-yum-to-honor-rpm-weak-dependency

Reply:
Arrived in `rpm` 4.13, CentOS 7 had 4.11, unless that's changed? And, `yum` doesn't support them, apparently...while dnf does. I guess CentOS 7 has `dnf`, but not sure it's new enough for Recommends?

Reply:
In that case, I'd suggest dropping `Recommends` for now.

Reply:
> In that case, I'd suggest dropping Recommends for now.

Absolutely no reason to drop it! If `rpm` version doesn't support it, `Recommends:` will simply be ignored. No damage.

Reply:
> Are you 100% sure about that? 

I am 1000% sure!

>  I recall that the RPM will be rejected on older systems if the Recommends: tag isn't recognized

CentOS 6 and CentOS 7 simply ignore it, but installation work:

**Example:**

<img width="1307" alt="image" src="https://user-images.githubusercontent.com/4426533/217021478-1294881c-20e5-4e8c-a5c8-19301fc7481a.png">

RHEL 8 and 9 treats `Recommends:` as expected and installs needed dependencies.

Which makes it work for all users.